providers/wlm/pod.go

```diff
@@ -0,0 +1,267 @@
+	sAPI "github.com/sylabs/wlm-operator/pkg/workload/api"
+	"github.com/virtual-kubelet/virtual-kubelet/vkubelet/api"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// CreatePod created new pod.
+// If a pod owner is a SlurmJob, it will start job on a Slurm cluster,
+// other pods will not be launched.
+func (p *Provider) CreatePod(ctx context.Context, pod *v1.Pod) error {
+	log.Printf("Create Pod %s", podName(pod.Namespace, pod.Name))
+
+	now := metav1.NewTime(time.Now())
+	pod.Status.StartTime = &now
+
+	job := newJob(*pod, p.wlmAPI, p.wlmClient)
+
+	if err := job.Start(); err != nil && err != errNotWlmJob {
+		return errors.Wrap(err, "Could not start job")
+	}
+
+	p.pods[podName(pod.Namespace, pod.Name)] = job
+
+	return nil
+}
+
+// UpdatePod updates pod.
+func (p *Provider) UpdatePod(ctx context.Context, pod *v1.Pod) error {
+	log.Printf("Update pod %s", podName(pod.Namespace, pod.Name))
+	pi, ok := p.pods[podName(pod.Namespace, pod.Name)]
+	if !ok {
+		return ErrPodNotFound
+	}
+	pi.pod = *pod
+
+	return nil
+}
+
+// DeletePod deletes pod.
+func (p *Provider) DeletePod(ctx context.Context, pod *v1.Pod) error {
+	log.Printf("Delete %s", podName(pod.Namespace, pod.Name))
+	pi := p.pods[podName(pod.Namespace, pod.Name)]
+	if err := pi.Cancel(); err != nil && err != errNotWlmJob {
+		return errors.Wrap(err, "Could not cancel job")
+	}
+
+	delete(p.pods, podName(pod.Namespace, pod.Name))
+	return nil
+}
+
+// GetPod returns a pod.
+func (p *Provider) GetPod(ctx context.Context, namespace, name string) (*v1.Pod, error) {
+	log.Printf("Get Pod %s", podName(namespace, name))
+	pj, ok := p.pods[podName(namespace, name)]
+	if !ok {
+		return nil, ErrPodNotFound
+	}
+
+	return &pj.pod, nil
+}
+
+// GetContainerLogs returns logs if requested pod is SlurmJob,
+// otherwise returns empty reader.
+func (p *Provider) GetContainerLogs(ctx context.Context, namespace, pName, containerName string, opts api.ContainerLogOpts) (io.ReadCloser, error) {
+	log.Printf("GetContainerLogs n:%s pod:%s containerName:%s", namespace, pName, containerName)
+
+	pi, ok := p.pods[podName(namespace, pName)]
+	if !ok {
+		return nil, ErrPodNotFound
+	}
+
+	logs, err := pi.Logs()
+	if err != nil {
+		if err != errNotWlmJob {
+			return nil, errors.Wrap(err, "Could not get logs")
+		}
+
+		return ioutil.NopCloser(strings.NewReader("")), nil
+	}
+
+	return ioutil.NopCloser(logs), nil
+}
+
+// RunInContainer wlm doesn't support it.
+func (p *Provider) RunInContainer(ctx context.Context, namespace, name, container string, cmd []string, attach api.AttachIO) error {
+	return ErrNotSupported
+}
+
+// GetPodStatus retrieves the status of a given pod by name.
+func (p *Provider) GetPodStatus(ctx context.Context, namespace, name string) (*v1.PodStatus, error) {
+	log.Printf("Get Pod Status namespace:%s name:%s", namespace, name)
+
+	status := &v1.PodStatus{
+		Phase:  v1.PodRunning,
+		HostIP: "1.2.3.4",
```

Comment:
Are those required?

Reply:
Not sure for now. Have created an issue for this https://github.com/sylabs/wlm-operator/issues/192